feat(routes): add tasks route

Reedgern opened the pull request.

Reedgern committed:
feat(routes): add todo route

YankinaDarya commented on src/_redux/todoTasks/reducer.ts:
```diff
@@ -0,0 +1,55 @@
+} from '@/_redux/todoTasks/actions';
+import { ITodoState } from './types';
+
+export const initialState: ITodoState = {
```
TodoStorageType

YankinaDarya commented on src/_redux/todoTasks/reducer.ts:
```diff
@@ -0,0 +1,55 @@
+        errors: [...state.errors, payload.error],
+      };
+    }
+    case CLEAR_ERRORS: {
```
лучше RESET_ERRORS

YankinaDarya commented on src/_redux/todoTasks/sagas.ts:
```diff
@@ -0,0 +1,78 @@
+  updateTaskRequest,
+} from '@/api/requests/todoTasks';
+
+function* fetchTodoTasksWorkerSaga() {
```
export

YankinaDarya commented on src/pages/todo/page/_components/tasksList/_components/taskCard/index.tsx:
```diff
@@ -0,0 +1,70 @@
+    }, []);
+
+    const onSubmitClick = useCallback(
+      (values) => {
```
тип values

YankinaDarya commented on src/pages/todo/page/_components/tasksList/_components/taskCard/index.tsx:
```diff
@@ -0,0 +1,70 @@
+export const TaskCard = memo(
+  ({ task: { id, description, isCompleted }, handleRemove }: PropsType) => {
+    const [isEditMode, setIsEditMode] = useState(false);
+    const dispatch = useDispatch();
```
это не нужно

YankinaDarya commented on src/pages/todo/page/_components/tasksList/_components/taskCard/index.tsx:
```diff
@@ -0,0 +1,70 @@
+
+    const onSubmitClick = useCallback(
+      (values) => {
+        dispatch(
```
это не нужно

YankinaDarya commented on src/pages/todo/page/_components/tasksList/_components/taskCard/index.tsx:
```diff
@@ -0,0 +1,70 @@
+    const onSubmitClick = useCallback(
+      (values) => {
+        dispatch(
+          updateTaskAction({
```
вызовы экшенов в контейнер-компоненте, которая завернута в коннект

YankinaDarya commented on src/pages/todo/page/_components/tasksList/index.tsx:
```diff
@@ -0,0 +1,26 @@
+const cn = classnames.bind(styles);
+
+type PropsType = {
+  tasks: ITodoState['tasks'];
```
`Array<TaskType>`

YankinaDarya commented on src/pages/todo/page/_components/tasksList/index.tsx:
```diff
@@ -0,0 +1,26 @@
+
+type PropsType = {
+  tasks: ITodoState['tasks'];
+  handleRemove: ({ id }: { id: string }) => void;
```
handleRemove: (id: string) => void;

YankinaDarya commented on src/pages/todo/page/index.tsx:
```diff
@@ -0,0 +1,82 @@
+} from '@/_redux/todoTasks';
+import { TasksList } from '@/pages/todo/page/_components/tasksList';
+import { TaskForm } from '@/pages/todo/page/_components/taskForm';
+import styles from './index.module.scss';
```
в контейнерной компоненте нет стилей, выделяем в отдельный view слой

YankinaDarya commented on src/pages/todo/page/index.tsx:
```diff
@@ -0,0 +1,82 @@
+
+const cn = classnames.bind(styles);
+
+const mapStateToProps = (state) => ({
```
тип у state и mapStateToProps с mapDispatchToProps в конец файла перед экспортом

YankinaDarya commented on src/pages/todo/page/index.tsx:
```diff
@@ -0,0 +1,82 @@
+  addNewTask: addNewTaskAction,
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
```
это не нужно

YankinaDarya commented on src/pages/todo/page/index.tsx:
```diff
@@ -0,0 +1,82 @@
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type PropsType = ConnectedProps<typeof connector>;
```
описать пропы объектом

YankinaDarya commented on src/pages/todo/page/index.tsx:
```diff
@@ -0,0 +1,82 @@
+    this.props.getTodoTasks();
+  }
+
+  onHandleSubmit = (values, form) => {
```
просто handleSubmit

YankinaDarya replied:
2 и более аргументов передаем объектом, и надо описать тип

YankinaDarya replied:
из формы юзается только reset, передавай только его, и лучше его вызвать только после того, как запрос успешно прошел

YankinaDarya commented on src/pages/todo/page/index.tsx:
```diff
@@ -0,0 +1,82 @@
+
+  render() {
+    return (
+      <div className={cn('wrapper')}>
```
всю разметку и стили в отдельный вью-компонент

YankinaDarya commented on src/pages/todo/page/_components/tasksList/index.tsx:
```diff
@@ -0,0 +1,26 @@
+};
+
+export const TasksList = memo(({ tasks, handleRemove }: PropsType) => {
+  if (tasks.length === 0) {
```
в юзмемо

YankinaDarya commented on src/pages/todo/page/index.tsx:
```diff
@@ -0,0 +1,82 @@
+          Создать новую таску:
+          <br />
+          <TaskForm
+            onCancel={(form) => form.reset()}
```
onCancel={reset}

Reedgern committed:
refactor(types): refactor action types and sagas

Reedgern committed:
fix(components): remove componentDidMount from TodoListPage component

Reedgern committed:
refactor(components): refactor structure

Reedgern committed:
refactor(server): update response body in tasks-controller

Reedgern committed:
feat(components): add notifications and form loading

mihanizm56 committed:
docs(make review comments): написал комментарии к коду проекта

написал комментарии к коду. Необходимо не только поправить текущие, но и пройтись по проекту на
предмет таких же недочётов

Reedgern committed:
refactor(src): fix review issues

Reedgern committed:
refactor(src): fix review issues

removed logic from reducer

Reedgern committed:
feat(components): move form into modal

Reedgern committed:
refactor(server): delete unused files

Reedgern committed:
feat(components): update taskCard styles

mihanizm56 committed:
build(components): review

Reedgern committed:
refactor(components): rewrite page component

Reedgern committed:
feat(components): replace initLoadManagerSaga with custom sagas

Reedgern committed:
refactor(redux): remove set-task-edit-mode saga

mihanizm56 committed:
build(configure-project): update

Reedgern committed:
feat(components): add translations

Reedgern committed:
fix(translations): fix translations

mihanizm56 committed:
Merge pull request #2 from Reedgern/refactor-1

refactor(redux): remove set-task-edit-mode saga
